docs/design/windows-design.md

```diff
@@ -108,94 +108,74 @@ The packets that are sent to/from the Windows host should be forwarded on this i
 is also a valid entry point into the OVS pipeline. A special ofport number 65534 (named as LOCAL) for the
 OVS bridge is used in OpenFlow spec.
 
-In the OVS `Classifier` table, new OpenFlow entries are needed to match the packets from this interface.
-There are two kinds of packets entering OVS pipeline from this interface:
-
- 1) the packet is sent from the Windows host to an external address
- 2) the packet is forwarded from the Windows host to the local Pod (e.g., this happens if the traffic is initiated
-  from a Pod to a NodePort Service, and the chosen Node is where the Pod is located)
-
-For 1, the packet is always output to the uplink interface directly. For 2, the packet is injected into
-the OVS pipeline and output to the backend Pod finally.
+In the OVS `Classifier` table, new OpenFlow entries are needed to match the packets from this interface. The
+packet entering OVS from this interface is output to the uplink interface directly.
 
 ### OVS uplink interface configuration
 
 After the OVS bridge is created, the original physical adapter is added to the OVS bridge as the uplink interface.
-The uplink interface is used to support traffic from Pods accessing external addresses. The packet is always
-output to the uplink interface if it is sent to an external address and is entering OVS from the bridge
-interface.
+The uplink interface is used to support traffic from Pods accessing the world outside current host.
 
 We should differentiate the traffic if it is entering OVS from the uplink interface in OVS `Classifier`
-table. There are two kinds of packets entering the OVS bridge from the uplink interface:
+table. In encap mode, the packets entering OVS from the uplink interface is output to the bridge interface directly.
+In noEncap mode, there are two kinds of packets entering OVS from the uplink interface:
 
- 1) reply traffic for the traffic that is sent from local Pods to external addresses,
- 2) traffic on the host network
+ 1) traffic that is sent to local Pods from Pod on a different Node,
+ 2) traffic that is sent to local Pods from a different Node according to the routing configuration,
+ 3) traffic on the host network
 
-For 1, the packets should enter the OVS pipeline and be re-written using SNAT information in connection tracking
-context first, and then processed by the OVS pipeline.
-For 2, the packets are output to the OVS bridge interface directly.
+For 1 and 2, the packet enters the OVS pipeline, and the `macRewriteMark` is set to ensure the destination MAC can be
+modified.
+For 3, the packet is output to the OVS bridge interface directly.
 
-### SNAT configuration
+The packet is always output to the uplink interface if it is entering OVS from the bridge interface. We
+also output the packet to the uplink interface in noEncap mode, if the destination is a Pod on a different Node,
```

Comment:
It seems some context is missing here. Is it for packets from the pod interface?

Reply:
updated.